src/Sylius/Behat/Context/Api/Admin/ManagingPaymentMethodsContext.php

```diff
@@ -0,0 +1,622 @@
+    }
+
+    /**
+     * @Then I should be notified that :element is required
+     */
+    public function iShouldBeNotifiedThatElementIsRequired(string $element): void
+    {
+        Assert::contains(
+            $this->responseChecker->getError($this->client->getLastResponse()),
+            sprintf('The type of the "%s" attribute must be "string", "NULL" given.', $element),
+        );
+    }
+
+    /**
+     * @Then I should be notified that I have to specify payment method :element
+     */
+    public function iShouldBeNotifiedThatINeedToSpecifyPaymentMethodName(string $element): void
+    {
+        Assert::contains(
+            $this->responseChecker->getError($this->client->getLastResponse()),
+            sprintf('%s: Please enter payment method %s.', $element, $element),
+        );
+    }
+
+    /**
+     * @Then I should be notified that I have to specify paypal :element
+     */
+    public function iShouldBeNotifiedThatIHaveToSpecifyPaypal($element)
+    {
+        Assert::same(
+            $this->createPage->getValidationMessage('paypal_' . $element),
+            sprintf('Please enter paypal %s.', $element),
+        );
+    }
+
+    /**
+     * @Then I should be notified that gateway name should contain only letters and underscores
+     */
+    public function iShouldBeNotifiedThatGatewayNameShouldContainOnlyLettersAndUnderscores()
+    {
+        Assert::same(
+            $this->createPage->getValidationMessage('gateway_name'),
+            'Gateway name should contain only letters and underscores.',
+        );
+    }
+
+    /**
+     * @Then the payment method with :element :value should not be added
+     */
+    public function thePaymentMethodWithElementValueShouldNotBeAdded($element, $value)
+    {
+        Assert::false(
+            $this->responseChecker->hasItemWithValue($this->client->index(Resources::PAYMENT_METHODS), $element, $value),
+            sprintf('Payment method with %s: %s exists', $element, $value),
+        );
+    }
+
+    /**
+     * @Then /^(this payment method) should still be named "([^"]+)"$/
+     */
+    public function thisShippingMethodNameShouldBe(PaymentMethodInterface $paymentMethod, $paymentMethodName)
+    {
+        Assert::true(
+            $this->responseChecker->hasItemWithValue($this->client->index(Resources::PAYMENT_METHODS), 'name', $paymentMethodName),
+            sprintf('Payment method with name %s does not exist', $paymentMethodName),
+        );
+    }
+
+    /**
+     * @param string $element
+     * @param string $expectedMessage
+     */
+    private function assertFieldValidationMessage($element, $expectedMessage)
+    {
+        /** @var CreatePageInterface|UpdatePageInterface $currentPage */
+        $currentPage = $this->currentPageResolver->getCurrentPageWithForm([$this->createPage, $this->updatePage]);
+
+        Assert::same($currentPage->getValidationMessage($element), $expectedMessage);
+    }
+
+    /**
+     * @Then the code field should be disabled
```

Comment:
```suggestion
     * @Then I should not be able to edit its code
```
and adjust the scenario, we have already done this with other resources